### PR TITLE
Add article and user talk page empty view

### PR DESCRIPTION
**Pull request by staykids**
**Phabricator:** https://phabricator.wikimedia.org/T311638 / https://phabricator.wikimedia.org/T310275 (partially)

### Notes
Adds the empty view to article and user talk pages. This also changes the `TalkPageAPIResponse` to consider an absence of topics as a successful call, because otherwise the decoding failure would propagate down the stack as an error. If my approach here is undesirable, I'm happy to change it.

### Test Steps
1. Open an empty user talk page (that is, one without any topics)
2. Confirm the empty view is shown, with the user's name in the text presented
3. Confirm the action button opens the new topic view
4. Open an empty article talk page (should be pretty easy to find on test wiki)
5. Confirm the empty view is shown
6. Confirm the action button opens the new topic view

### CODE REVIEW

**Review by tonisevener — COMMENTED** [review]
Looking great! Just some thoughts to open up some discussion, but I can approve as-is if you feel strongly about it.

File: Wikipedia/Code/TalkPageViewController.swift [hunk 1]
```diff
@@ -400,6 +402,11 @@ class TalkPageViewController: ViewController {
 extension TalkPageViewController: UICollectionViewDelegate, UICollectionViewDataSource {
     
     func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
```
[the remark this answers is not in the document]
Comment (staykids) [reply Wikipedia/Code/TalkPageViewController.swift:404]: I totally agree, it's uncomfortably placed. Yeah there're definitely pros and cons either way: a pro of this approach is that the logic that handles empty state presentation is located in a somewhat logical location, is triggered automatically on data updates without developer intervention, and doesn't require remembering to call a separate method in fetch completion blocks. But an obvious con like you mention is that this collection view delegate method is now behaviorally overloaded and has side effects. I think probably ideally the fetch method calls some kind of delegate method that automatically handles this work without developer intervention.

But I'll just start with the suggestion you have about separating this work out and returning this collection view delegate method to being simplified and single purposed.

File: Wikipedia/Code/TalkPageViewController.swift [hunk 2]
```diff
@@ -400,6 +402,11 @@ class TalkPageViewController: ViewController {
...
+        if viewModel.hasPerformedInitialFetch {
+            talkPageView.updateEmptyView(visible: viewModel.topics.count == 0)
+            scrollView = viewModel.topics.count == 0 ? talkPageView.emptyView.scrollView : talkPageView.collectionView
```
[the remark this answers is not in the document]
Comment (staykids) [reply Wikipedia/Code/TalkPageViewController.swift:407]: Great catch - will update along with collection view delegate method refactor.

**Comment by staykids** [comment]
Updated and ready for another look!

File: Wikipedia/Code/TalkPageEmptyView.swift [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+        static let userBody = WMFLocalizedString("talk-pages-empty-view-body-user", value: "Talk pages are where people discuss how to make content on Wikipedia the best that it can be. Start a new discussion to connect and collaborate with %1$@. What you say here will be public for others to see.", comment: "Text displayed in user talk pages when no topics are available. %1$@ is replaced with a username.")
+        static let startDiscussion = WMFLocalizedString("talk-pages-empty-view-button-user-start-discussion", value: "Start a discussion", comment: "Button text displayed in user talk pages when no topics are available.")
+        static let addTopic = WMFLocalizedString("talk-pages-empty-view-button-article-add-topic", value: "Add a new topic", comment: "Button text displayed in article talk pages when no topics are available.")
+    }
+
+    // MARK: - Properties
+
+    fileprivate var headerLabelPrimaryFont = UIFont.wmf_scaledSystemFont(forTextStyle: .title1, weight: .semibold, size: 28)
```
[the remark this answers is not in the document]
Comment (tonisevener) [reply Wikipedia/Code/TalkPageEmptyView.swift:20]: @staykids Makes sense, thanks for clearing that up! I missed that we weren't able to customize sizing in the old styles. Also, good breakdown of the choice we have in the future. I'm looking forward to us tackling this class again.

File: Wikipedia/Code/TalkPageViewController.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (tonisevener) [reply Wikipedia/Code/TalkPageViewController.swift]: Looks good, thanks!

**Review by tonisevener — APPROVED** [review]

**Pull request merged**